main.js

```diff
@@ -194,18 +203,81 @@ function createWindow () {
   });
 }
 
+function createTrayIcon() {
+  logger.info('Creating tray icon');
+
+  // TODO: better to not use a hardcoded icon file
+  tray = new Tray(path.join(__dirname, '/images/icon_48.png'));
+
+  trayContextMenu = Menu.buildFromTemplate([
+      {id: 'show', label: locale.messages.show.message, visible: !mainWindow.isVisible(), click: () => {showWindow();}},
+      {id: 'hide', label: locale.messages.hide.message, visible: mainWindow.isVisible(), click: () => {hideWindow();}},
+      {id: 'quit', label: locale.messages.quit.message, click: () => {app.quit();}}
+  ]);
+
+  // NOTE: a function equivalent to the following will become part of the
+  // prototype of Menu in a future release of electron (atm it is in 1.8.1-beta),
+  // so this definition can be removed in the future, when that version of electron
+  // will be released and stable. For details, please see:
+  // https://github.com/electron/electron/releases/tag/v1.8.1
+  // https://github.com/electron/electron/pull/10618
+  trayContextMenu.getMenuItemById = function (id) {
+    return this.items.find(item => item.id === id) || null;
```

Comment:
The `|| null` here seems unecessary. Below, where this is used, there's no check before setting its `visible` property, so it will crash. Maybe the right thing to do here is to throw if we can't find any menu item with the provided id?

Reply:
I think throwing an exception there would be the cleanest way. But the method implemented in electron does not do that, it just returns null, so if we want to be able to replace this function with it we may need to do null checks (that I added where needed, thank you for pointing it out).
Otherwise we can do differently and not rely on the (future) electron's API for it (after all, this is a very short piece of code and it is called just twice).